### ISSUE TITLE
Some queries are broken since v1.34.5: cannot find metricName by metricID

### ISSUE THREAD

**Issue by glebsam** [VictoriaMetrics/VictoriaMetrics#416]
**Describe the bug**
Since VictoriaMetrics `v1.34.5` cannot be executed query like the following:
```
{__name__="container_cpu_usage_seconds_total", container_label_com_amazonaws_ecs_container_name=~".+", ec2_tag_project=~".+"}
```

I use single-node installation, `v1.34.4` works fine, the issue reproduced with `v1.34.5`, `v1.34.6` and `v1.34.7`. After execution I getting the following in logs:

`2020-04-13T11:15:57.132Z	error	VictoriaMetrics/app/vmselect/main.go:236	error in "/api/v1/series?match[]=container_cpu_usage_seconds_total%7Bcontainer_label_com_amazonaws_ecs_container_name%3D~%22.%2B%22%2C%20ec2_tag_project%3D~%22.%2B%22%7D&start=1586733358&end=1586776558": cannot fetch data for "MinTimestamp=2020-04-12 23:15:58 +0000 UTC, MaxTimestamp=2020-04-13 11:15:58 +0000 UTC, TagFilters=[\n{Key=\"\", Value=\"container_cpu_usage_seconds_total\", IsNegative: false, IsRegexp: false}{Key=\"container_label_com_amazonaws_ecs_container_name\", Value=\".+\", IsNegative: false, IsRegexp: true}{Key=\"ec2_tag_project\", Value=\".+\", IsNegative: false, IsRegexp: true}\n]": search error after reading 0 data blocks: error when searching tsids for tfss ["{__name__=\"container_cpu_usage_seconds_total\", container_label_com_amazonaws_ecs_container_name=~\".+\", ec2_tag_project=~\".+\"}"]: cannot search for metricIDs for 2020-04-13 00:00:00 +0000 UTC: cannot find metricName by metricID 1582417212213420669: EOF`

So, human-readable query is: `/api/v1/series?match[]=container_cpu_usage_seconds_total{container_label_com_amazonaws_ecs_container_name=~".+", ec2_tag_project=~".+"}&start=1586733358&end=1586776558`

**To Reproduce**
1. Run docker image with argument `-storageDataPath=/data`. 
2. Try the query mentioned above (I use Explore feature in Grafana, `curl` also reproduces). 

**Expected behavior**
The query executed successfully. 

**Version**
```
❯ docker run --rm -it victoriametrics/victoria-metrics:latest --version
victoria-metrics-20200405-113850-tags-v1.34.7-0-g69962a70
```

**Used command-line flags**
`-storageDataPath=/data`

**Additional information**

Telegram channel [has a mention](https://t.me/VictoriaMetrics_ru1/9739) of the similar issue, but that time it was solved by cleaning `/data/cache` directory. I [had no luck with this solution](https://t.me/VictoriaMetrics_ru1/10532), though.

**Comment by valyala** [comment]
@glebsam , thanks for the bug report! It should be fixed in the commit https://github.com/VictoriaMetrics/VictoriaMetrics/commit/c1de3f67b4a1c8b25ad70a41e9c5714c260d78d5 . Could you build VictoriaMetrics from this commit and verify whether the issue is fixed?

See [these docs on how to build VictoriaMetrics from sources](https://github.com/VictoriaMetrics/VictoriaMetrics/blob/master/README.md#how-to-build-from-sources).

**Comment by glebsam** [comment]
@valyala unfortunately, I cannot reproduce the issue on `v1.34.7` anymore, the same query works fine now.

**Comment by valyala** [comment]
The issue should be reproduced only when new time series are ingested into VictoriaMetrics during the query. VictoriaMetrics adds multiple entries into inverted index when it encounters new time series.  These entries become visible for search only after being flushed to disk (this usually takes up to a second). The bug is triggered when only a part of these entries for new time series are flushed to disk, while another part remained in the in-memory buffer (it should be flushed in the next second). In this case VictoriaMetrics cannot find `metricID->metricName` entry in the inverted index, since it wasn't flushed to disk yet.

**Comment by valyala** [comment]
The bug should be fixed in [v1.34.8](https://github.com/VictoriaMetrics/VictoriaMetrics/releases). Closing the issue as fixed.
